language/Modules.md

```diff
@@ -55,7 +55,49 @@ import B (runFoo)
 
 ## Qualified Imports
   
-Modules can also be imported qualified, which means that their names will not be brought directly into scope, but rather, aliased to a different module name. This can be helpful when avoiding naming conflicts:
+Modules can also be imported qualified, which means that their names will not be brought directly into scope, but rather, aliased as a different name.
+
+Following are some situations in which qualified imports are quite useful.
+
+### Using generically-named functions
+
+``` purescript
+module Main where
+
+import Data.Map as Map
+import Data.List as List
+import Data.Set as Set
+
+a :: Map Int String
+a = Map.fromFoldable [ Tuple 1 "a" ]
+
+b :: List Int
+b = List.fromFoldable [ 1 ]
+
+c :: Set Int
+c = Set.fromFoldable [ 1 ]
+```
+
+Several data structure modules have a `fromFoldable` function which can be used to create an instance of that data structure from any other `Foldable` data structure. We can say the `fromFoldable` function name is overloaded by different modules. To clarify which `fromFoldable` function is being used, we can import that module's functions under a qualified name and use it qualified, like `Set.fromFoldable`.
```

Comment:
If we're importing all of `Data.Map`, `Data.List`, and `Data.Set` in the example, this really becomes an example of using qualified imports to avoid naming conflicts. I think we should only be using one `fromFoldable` if we want to make the point that it improves readability. Also, I'm not sure we need a whole module for this example; we can just say that if you're reading code and you see `fromFoldable`, you might refer to the import list to see where it comes from.

Also, I don't think the "we can say the `fromFoldable` function name is overloaded by different modules" really adds anything here; the previous sentence is sufficient, I think.